app/sgqlc.py

```diff
@@ -38,87 +51,104 @@ def convert_query(self, item, query):
         if "null" in snake_case_query:
             snake_case_query = re.sub(
                 '[,]? [_a-z]+: null', '', snake_case_query)
-        # Update the filter query node name
-        if "filter" in item.node:  # query situation
-            snake_case_query = re.sub('_filter', '', snake_case_query)
-            item.node = re.sub('_filter', '', item.node)
-        if type(item.search) == dict:  # pagination situation
+        # Either pagination or experiment node query
+        if "experiment" in item.node:
             snake_case_query = self.update_manifests_information(
                 snake_case_query)
-            # Only pagination will need to add count field
-            snake_case_query = self.add_count_field(item, snake_case_query)
+            # Only pagination will need count field
+            if type(item.search) == dict:
+                snake_case_query = self.add_count_field(item, snake_case_query)
+        snake_case_query, item.node = self.remove_node_suffix(
+            item.node, snake_case_query)
         return "{" + snake_case_query + "}"
 
-    # if the node name contains "_filter",
-    # the query generator will only be used for /graphql/pagination API
-    # else is for /graphql/query API,
-    # this will fetch all the fields that Gen3 metadata has
+    # generated query will fetch all the fields that Gen3 metadata has
     def generate_query(self, item):
         query = Operation(Query)
-        if item.node == "experiment":
+        # FILTER
+        # if the node name contains "_filter",
+        # the query generator will be used for /filter/ and /graphql/pagination API
+        if item.node == "dataset_description_filter":
             return self.convert_query(
                 item,
-                query.experiment(
-                    first=item.limit,
-                    offset=(item.page-1)*item.limit,
-                    submitter_id=item.filter["submitter_id"] if "submitter_id" in item.filter else None
+                query.datasetDescriptionFilter(
+                    first=0,
+                    offset=0,
+                    # study_organ_system=item.filter["study_organ_system"] if "study_organ_system" in item.filter else None
                 )
             )
-        elif item.node == "dataset_description":
+        elif item.node == "manifest_filter":
             return self.convert_query(
                 item,
-                query.datasetDescription(
-                    first=item.limit,
-                    offset=(item.page-1)*item.limit,
-                    submitter_id=item.filter["submitter_id"] if "submitter_id" in item.filter else None
+                query.manifestFilter(
+                    first=0,
+                    offset=0,
+                    additional_types=item.filter["additional_types"] if "additional_types" in item.filter else None
```

Comment:
A get function for dictionary is probably clearer - 
e.g. `additional_types=item.filter.get("additional_types", None)`

Reply:
Sure, will update it now.